src/test/xmeta/test_symbol_table.cpp

```diff
@@ -0,0 +1,92 @@
+#include "pch.h"
+
+#include <algorithm>
+#include <iostream>
+#include <string_view>
+
+#include "antlr4-runtime.h"
+#include "symbol_table.h"
+#include "XlangParser.h"
+#include "XlangLexer.h"
+#include "ast_to_st_listener.h"
+#include "helpers.h"
+
+using namespace antlr4;
+using namespace xlang::xmeta;
+
+xmeta_symbol_table st{ "" };
+
+TEST_CASE("Duplicate Namespaces")
+{
+    std::string test_idl =
+        "namespace n { } \
+        namespace n { }";
+
+    st.reset("dup_ns.idl");
+    ast_to_st_listener listener{ st };
+    REQUIRE(setup_and_run_parser(test_idl, listener) == 0);
+
+    auto namespaces = st.get_namespaces();
+    auto it = namespaces.find("n");
+    REQUIRE(it != namespaces.end());
+    auto ns = it->second;
+    REQUIRE(ns->get_namespace_bodies().size() == 2);
+    REQUIRE(ns->get_namespace_bodies()[0]->get_containing_namespace() == ns);
+    REQUIRE(ns->get_namespace_bodies()[1]->get_containing_namespace() == ns);
+}
+
+TEST_CASE("Enum test")
+{
+    std::string test_idl =
+        "namespace n \
+        { \
+            enum e \
+            { \
+                e_member_1, \
+                e_member_2 = 3, \
+                e_member_3, \
+                e_member_4 = e_member_5, \
+                e_member_5 = 0x21 \
+            } \
+        } ";
+
+    ANTLRInputStream input(test_idl);
+    XlangLexer lexer(&input);
+    CommonTokenStream tokens(&lexer);
+    XlangParser parser(&tokens);
+
+    auto tree = parser.xlang();
+    st.reset("ns_members.idl");
+    ast_to_st_listener listener{ st };
+    tree::ParseTreeWalker::DEFAULT.walk(&listener, tree);
```

Comment:
Why don't you use setup_and_run_parser here? You can disable error reporting here but setting the last param to true if you know there are gong to be errors. 

setup_and_run_parser(test_idl, listener, true)